packages/teuchos/core/src/Teuchos_TestingHelpers.hpp

```diff
@@ -258,6 +263,30 @@ bool compareFloatingArrays(
   Teuchos::FancyOStream &out
   );
 
+/** \brief Compare if two array objects are the same up to an absolute
+ * tolerance: elements <tt>a1[i]</tt> and <tt>a2[i]</tt> are considered
+ * the same if <tt>|a1[i]-a2[i]| <= tol</tt>.
+ *
+ * This function works with any two array-like objects
+ * with the same element types. <tt>Array1</tt> and <tt>Array2</tt> must have
+ * <tt>size()</tt> and <tt>operator[](i)</tt> methods defined.
+ * <tt>Teuchos::ScalarTraits</tt> must also have a specialization
+ * for the element type.
+ *
+ * \returns Returns <tt>true</tt> if <tt>a1.size() == a2.size()</tt> and
+ *   <tt>|a1[i]-a2[i]| <= tol</tt> for all <tt>0 <= i < a1.size()</tt>.
+ *   Otherwise returns <tt>false</tt>.
+ *
+ * \ingroup teuchos_testing_grp
+ */
+template<class Array1, class Array2, class ScalarMag>
+bool compareFloatingArraysAbsolute(
```

Comment:
Technically, we could unit test this function to ensure that it behaves as we expect.  But I see that the current function `Teuchos::compareFloatingArrays()` (which I added 16 years ago in commit 23ce08e7e6d 😊) did not have a unit test for this so no big deal.   (But this function is easy enough to manually check that it works as it should while writing unit tests so not much risk there.  In fact, I aways start with a failing test, look at the output, and then make the test pass, which tests the unit testing infrastructure.)